Update events docs for pageview and pageleave (#6121)
PostHog/posthog.com: contents/docs/data/events.mdx

You can also click each user to view a their entire event history.

## Default events and properties

You may notice that some events or properties are prefixed with `$` e.g., `$pageview`, `$feature_flag_called`, `$current_url`, or `$os_version`. This indicates that these are default PostHog events or properties.

### Pageview events

> By default, pageview events are automatically sent when using the [JavaScript snippet or SDK](/docs/libraries/js). You can disable this by setting `capture_pageview: false` in the [config](/docs/libraries/js#config).

Reviewer: ```suggestion
`$pageview` events are automatically sent when using the [JavaScript snippet or SDK](/docs/libraries/js). You can disable this by setting `capture_pageview: false` in the initialization [config](/docs/libraries/js#config).
```

> You can manually send pageview events by calling `posthog.capture('$pageview')`. You'll need to do this in single-page applications like [React](/docs/libraries/react), [Next.js](/docs/libraries/next-js#pages-router) or [Vue](/docs/libraries/vue-js#capturing-page-views), since otherwise a pageview event will only be sent for the first page.

Reviewer: ```suggestion
You can manually send pageview events by calling `posthog.capture('$pageview')`. This is necessary for single-page applications like [React](/docs/libraries/react), [Next.js](/docs/libraries/next-js#pages-router) or [Vue](/docs/libraries/vue-js#capturing-page-views), since otherwise a pageview event is only sent for the first page.
```

### Pageleave events

> Similar to pageview, pageleave events are automatically sent and be disabled by setting `capture_pageview: false` in the [config](/docs/libraries/js#config). For single-page applications, you'll also need to manually send these events using `posthog.capture('$pageleave')`.

Reviewer 2: Correction of the config option:
```suggestion
Similar to pageview, pageleave events are automatically sent and be disabled by setting `capture_pageleave: false` in the [config](/docs/libraries/js#config). For single-page applications, you'll also need to manually send these events using `posthog.capture('$pageleave')`.
```

Also I don't think this is true. SPAs don't need to implement their own pageleave as that happens when the browser unloads which is unrelated to the SPA.

Author: Nice catch! 
Re the SPA: In our [NextJs](https://posthog.com/docs/libraries/next-js#pages-router), or [Vue](https://posthog.com/docs/libraries/vue-js#capturing-page-views), you'll that you need to add this, otherwise they wont be logged? 

Unless you're talking about pageleave specifically. Im not sure why this would be different than pageview?

Reviewer 2: Pageview is only automatically triggered when the page loads (i.e. a full browser load) and pageleave when it unloads.

SPAs do routing in between the browser unloading, hence the need for the separate `pageview` command.
Pageleave doesn't really _do_ much so saying it needs to be implemented manually is (I would argue) a bit confusing. Its only use case is for tracking the unloading of the browser.

Author: Can you elaborate what you mean by "Pageleave doesn't really do much" ?

My concern is that it's misleading if we have an event called "page leave" and our users are expecting to it log when a user navigates away from page, but its not doings so for SPAs.

Reviewer 2: So there is no "special use case" for $pageleave. If it were renamed to `$page_unload` I think it would make more sense. 

With a normal website, when it is loaded we immediately capture a `$pageview`. We then try and listen for when the page "unloads" and we track a `$pageleave`.

With a SPA this is more tricky as there are multiple in-between "pageviews" as you navigate around the product. An additional "pageleave" would be pointless as it is fired at the same time and doesn't have any special properties.

The only useful thing for `$pageleave` as a dedicated event is if you wanted to know "time spent on page" which for SPAs could be handled via the in-between pageview events (if we added some special properties) and then `pageleave` (i.e. `page_unload` covers the case where the person closes the browser or clicks a link away from the SPA.

> Similar to pageview, pageleave events are automatically sent and be disabled by setting `capture_pageview: false` in the [config](/docs/libraries/js#config). For single-page applications, you'll also need to manually send these events using `posthog.capture('$pageleave')`.

Reviewer: ```suggestion
Similar to pageview, `$pageleave` events are automatically sent. You can disable them by setting `capture_pageleave: false` in the [config](/docs/libraries/js#config). For single-page applications, you'll also need to manually send these events using `posthog.capture('$pageleave')`.
```
